src/Extension/FluentSiteTreeExtension.php

```diff
@@ -95,4 +101,150 @@ public function updateLink(&$link, &$action, &$relativeLink)
+            return;
+        }
+
+        // We don't need to add a status warning if a version of this Page has already been published for this Locale.
+        if ($this->isPublishedInLocale()) {
+            return;
+        }
+
+        if (!$this->isDraftedInLocale()) {
+            $message = _t(
+                __CLASS__ . 'LOCALESTATUSINHERITED',
+                'Content for this Page is being inherited from another Locale. If you wish you make an independent copy
+                of this Page, please use one of the "Copy" actions provided.'
+            );
+        } else {
+            $message = _t(
+                __CLASS__ . 'LOCALESTATUSDRAFT',
+                'A draft has been created for this Locale, however, published content is still being inherited from
+                another Locale. To publish this content for this Locale, use the "Save & publish" action provided.'
+            );
+        }
+
+        $fields->unshift(
+            LiteralField::create(
+                'LocaleStatusMessage',
+                sprintf(
+                    '<p class="message notice">%s</p>',
+                    $message
+                )
+            )
+        );
+    }
+
+    /**
+     * @param FieldList $actions
+     */
+    protected function updateSavePublishActions(FieldList $actions)
+    {
+        /** @var \SilverStripe\Forms\CompositeField $majorActions */
+        $majorActions = $actions->fieldByName('MajorActions');
+
+        // If another extension has removed this CompositeField then we don't need to update them.
+        if ($majorActions === null) {
+            return;
+        }
+
+        // There's no need to update actions in these ways if the Page has previously been published in this Locale.
+        if ($this->isPublishedInLocale()) {
+            return;
+        }
+
+        $isDraftedInLocale = $this->isDraftedInLocale();
+        $actionSave = $majorActions->getChildren()->fieldByName('action_save');
+        $actionPublish = $majorActions->getChildren()->fieldByName('action_publish');
+
+        // Make sure no other extensions have removed this field.
+        if ($actionSave !== null) {
+            // Check that the Page doesn't have a current draft.
+            if (!$isDraftedInLocale) {
+                $actionSave->addExtraClass('btn-primary font-icon-save');
+                $actionSave->setTitle(_t(__CLASS__ . '.LOCALECOPYTODRAFT', 'Copy to draft'));
+                $actionSave->removeExtraClass('btn-outline-primary font-icon-tick');
+            }
+        }
+
+        // Make sure no other extensions have removed this field.
+        if ($actionPublish !== null) {
+            $actionPublish->addExtraClass('btn-primary font-icon-rocket');
+            $actionPublish->removeExtraClass('btn-outline-primary font-icon-tick');
+
+            if ($isDraftedInLocale) {
+                $actionPublish->setTitle(_t(SiteTree::class . '.BUTTONSAVEPUBLISH', 'Save & publish'));
```

Comment:
Unfortunately we can't use `::class` in translations yet, will need to be a full class string

Reply:
Thanks, @robbieaverill. Just to confirm, you would be ok with me switching out all usages of `::class` with a string EG: `"SilverStripe\CMS\Model\SiteTree"`?

Reply:
In translation calls only, yeah :-)